app/controllers/searches_controller.rb

```diff
@@ -32,6 +50,18 @@ def search
   end
 
   private
+    # TODO this function copied from books_controller. put in a helper class
+    # given an identifier string, get the language ID
+    # only returns an ID if the verbatim search returns exactly one result
+    def id_check(input_lang)
```

Comment:
This function should be in a helper class, but I wasn't familiar with where this class would go or how to add a dependency to it.